### PR TITLE
Chat testing

### PR DESCRIPTION
**Pull request by KmBarnett**
# Description :: User Story

## Type of change

- [x] New Feature
- [ ] Bug Fix
- [x] Refactor
- [ ] Breaking Change

## Notes
This PR adds some testing and socket handling
## Test results
passing

### CODE REVIEW

File: src/Components/ChatContainer/ChatContainer.js [hunk 1]
```diff
@@ -12,34 +12,27 @@ const endPoint = "http://localhost:5000";
...
-      setRoomData(data)
+      setRoomDetails(data)
       socket.emit('recived', true)
     });
-  },[roomData])
+  },[roomDetails])
 
   const leaveChat = () => {
-    socket.emit('leave', roomData)
+    socket.emit('leave', roomDetails)
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.js]: necessary?

File: src/Components/ChatContainer/ChatContainer.test.js [hunk 2]
```diff
@@ -80,6 +85,29 @@ describe("<ChatContainer />", () => {
...
+
+    socket.on('message', function (message) {
+      socket.emit('message', message)
+     });
+
+    fireEvent.change(messageInput, { target: { value: 'Test' } })
+    fireEvent.click(messageSend)
+
+    socket.on('recived', function (recived) {
+        expect(recived).toBe(true);
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.test.js:105]: *received

File: src/Components/ChatContainer/ChatContainer.js [hunk 3]
```diff
@@ -25,9 +25,8 @@ function ChatContainer({ groupCode, roomCode, name }) {
...
   useEffect(() => {
-    socket.on("message",function(msg, fun) {


+    socket.on("message",function(msg) {


       getMessages(msg);
-      console.log(fun);;
       socket.emit('recived', true)
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.js:23]: *received

File: src/Components/ChatContainer/ChatContainer.test.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ import rootReducer from '../../reducers';
 
 const testStore = createStore(rootReducer);
 
-let socket = new SocketMock();
+let socket
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.test.js:17]: what is this doing? how does it function if not set to any value?

File: src/Components/ChatContainer/ChatContainer.js [hunk 5]
```diff
@@ -9,11 +9,10 @@ import { Redirect } from 'react-router';
...


 
-function ChatContainer({ groupCode, roomCode }) {
+function ChatContainer({ groupCode, roomCode, name }) {
   const socket = io.connect(`${endPoint}`);
   const [messages, setMessages] = useState([]);
-  const [groupDetails, setGroupDetails] = useState({});
-  const [room, setRoom] = useState('');
+  const [roomData, setRoomData] = useState('');
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.js]: room and group details are contained w/in roomData?

File: src/Components/ChatContainer/ChatContainer.js [hunk 6]
```diff
@@ -26,28 +25,26 @@ function ChatContainer({ groupCode, roomCode }) {
...
-    socket.on("message",function(msg) {


-      getMessages(msg);

+    socket.on("message",function(msg, fun) {


+      getMessages(msg);
+      console.log(fun);;
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.js]: why this instead of above?

File: src/Components/ChatContainer/ChatContainer.js [hunk 7]
```diff
@@ -26,28 +25,26 @@ function ChatContainer({ groupCode, roomCode }) {
...
-    });

-
-  },[messages.length, groupDetails])
+  useEffect(() => {
+    socket.on("join_room",function(data) {


+      setRoomData(data)
+      socket.emit('recived', true)
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.js:30]: *

File: src/Components/ChatContainer/ChatContainer.test.js [hunk 8]
```diff
@@ -13,7 +14,10 @@ import rootReducer from '../../reducers';
 
 const testStore = createStore(rootReducer);
 
- let socket = new SocketMock();
+let socket = new SocketMock();
+
+const rules = [{id: uniqid(), rule: 'No Swearing'},{id: uniqid(), rule: 'No Nudity'}, {id: uniqid(), rule: 'Test'}]
```
Comment (Yetidancer) [inline src/Components/ChatContainer/ChatContainer.test.js:19]: why do rules need ids?

File: src/Reducers/name.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export const name = (state = '', action) => {
+  switch (action.type) {
+    case 'ADD_NAME':
+        return action.name
```
Comment (Yetidancer) [inline src/Reducers/name.js:4]: how does action function here?

File: src/Components/RoomForm/RoomForm.js [hunk 10]
```diff
@@ -77,9 +76,6 @@ function RoomForm({getCircleCode}) {
           onChange={(e) => { setRoomname(e.target.value) }}
           />
         </section>
-        <section className='input-container'>
-          <label>Group Code:</label> <h2>{ roomID }</h2>
-        </section>
```
Comment (Yetidancer) [inline src/Components/RoomForm/RoomForm.js:82]: just going with group name now?

File: src/Components/ChatContainer/ChatContainer.test.js (hunk 8 above)
Comment (Yetidancer) [reply src/Components/ChatContainer/ChatContainer.test.js:19]: nvm

File: src/Components/ChatContainer/ChatContainer.js (hunk 1 above)
Comment (KmBarnett) [reply src/Components/ChatContainer/ChatContainer.js]: Both of us on separate branches declared a roomDetails/roomData state so I just got rid of one?

File: src/Components/ChatContainer/ChatContainer.test.js (hunk 4 above)
Comment (KmBarnett) [reply src/Components/ChatContainer/ChatContainer.test.js:17]: So here I am declaring a variable and then later on in the before each I am assigning it

File: src/Components/ChatContainer/ChatContainer.js (hunk 5 above)
Comment (KmBarnett) [reply src/Components/ChatContainer/ChatContainer.js]: Room data was replaced with room details, and groupDetails isn't apart of the MVP for right now

File: src/Components/ChatContainer/ChatContainer.js (hunk 6 above)
Comment (KmBarnett) [reply src/Components/ChatContainer/ChatContainer.js]: This was an old change and the only difference is fun where I was playing around with what could comback

File: src/Components/RoomForm/RoomForm.js (hunk 10 above)
Comment (KmBarnett) [reply src/Components/RoomForm/RoomForm.js:82]: Yeah as we discussed with Ryan he doesn't want to start with the group code from the get-go

**Pull request merged**